lib/Common/DataStructures/QuickSort.h

```diff
@@ -0,0 +1,53 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#pragma once
+namespace JsUtil
+{
+    template <class T, typename Comparer>
+    class QuickSort
+    {
+    public:
+        static void Sort(T* low, T* high, Comparer comparer, void* context)
+        {
+            if ((low == NULL)||(high == NULL))
+            {
+                return;
+            }
+            if (high > low)
+            {
+                T* pivot = Partition(low, high, comparer, context);
+                Sort(low, pivot-1, comparer, context);
+                Sort(pivot+1, high, comparer, context);
+            }
+        }
+
+    private:
+        static T* Partition(T* l, T* r, Comparer comparer, void* context)
```

Comment:
This quicksort implementation performs poorly on already ordered array or array with large amount of equal elements. Good for now, will need to consider future improvements.